__tests__/app.test.js

```diff
@@ -22,3 +22,48 @@ describe('/api/topics', () => {
 			})
 	})
 })
+
+describe('/api', () => {
+	test('GET:200 esponds with an object describing all the available endpoints on your API', () => {
+		return request(app)
+			.get('/api')
+			.expect(200)
+			.then((response) => {
+				const endpoints = response.text
+				expect(typeof endpoints).toBe('string')
+
+				const parsedEndpoints = JSON.parse(endpoints)
+				expect(typeof parsedEndpoints).toBe('object')
+
+				for (let endpoint in parsedEndpoints) {
+					expect(parsedEndpoints[endpoint]).toHaveProperty(
+						'description',
+						expect.any(String)
+					)
+
+					if (endpoint.startsWith('GET /api/')) {
+						expect(parsedEndpoints[endpoint]).toHaveProperty(
+							'queries',
+							expect.any(Object)
+						)
+						expect(parsedEndpoints[endpoint]).toHaveProperty(
+							'exampleResponse',
+							expect.any(Object)
+						)
+					}
+
+					if (endpoint.startsWith('POST' || 'PATCH')) {
+						expect(parsedEndpoints[endpoint]).toHaveProperty(
+							'requestBodyFormat',
+							expect.any(Object)
+						)
+
+						expect(parsedEndpoints[endpoint]).toHaveProperty(
+							'exampleResponse',
+							expect.any(Object)
+						)
+					}
+				}
+			})
+	})
+})
```

Comment:
Nice work with this test! However, and this might annoy you, you could just require in the json file and check that the response looks the same 😄 

You can keep it as is or change it, up to you!

Reply:
The takeaway lesson here is work smart, not hard. I will change it and re submit